From aec2f3a2b7fe3bbcd48811c805d567884f319018 Mon Sep 17 00:00:00 2001
From: Jan Kiszka <jan.kiszka@siemens.com>
Date: Sun, 30 Dec 2018 12:59:35 +0100
Subject: [PATCH] pwrseq_simple: Workaround for missing device tree entry on
 hikey

We need a small power-on delay, but we use the firmware device tree with
Isar/jailhouse-images. Inject the delta via the driver until a new
firmware with the appropriate device tree change is available.

Not for upstream.

Signed-off-by: Jan Kiszka <jan.kiszka@siemens.com>
---
 drivers/mmc/core/pwrseq_simple.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/mmc/core/pwrseq_simple.c b/drivers/mmc/core/pwrseq_simple.c
index a8b9fee4d62a..f0ed2e92ffa6 100644
--- a/drivers/mmc/core/pwrseq_simple.c
+++ b/drivers/mmc/core/pwrseq_simple.c
@@ -126,6 +126,9 @@ static int mmc_pwrseq_simple_probe(struct platform_device *pdev)
 		return PTR_ERR(pwrseq->reset_gpios);
 	}
 
+	if (strcmp(pdev->name, "wl1835-pwrseq") == 0)
+		pwrseq->post_power_on_delay_ms = 1;
+
 	device_property_read_u32(dev, "post-power-on-delay-ms",
 				 &pwrseq->post_power_on_delay_ms);
 	device_property_read_u32(dev, "power-off-delay-us",
-- 
2.16.4

